ast/Trees.cc

```diff
@@ -1108,8 +1108,13 @@ ExpressionPtr Send::withNewBody(core::LocOffsets loc, ExpressionPtr recv, core::
 string Cast::toStringWithTabs(const core::GlobalState &gs, int tabs) const {
     fmt::memory_buffer buf;
     fmt::format_to(std::back_inserter(buf), "T.{}", this->cast.toString(gs));
-    fmt::format_to(std::back_inserter(buf), "({}, {})", this->arg.toStringWithTabs(gs, tabs),
-                   this->type.toStringWithTabs(gs, tabs));
+    fmt::format_to(std::back_inserter(buf), "({}, ", this->arg.toStringWithTabs(gs, tabs));
+    if (this->type != core::Types::todo()) {
+        fmt::format_to(std::back_inserter(buf), "{}", this->type.toStringWithTabs(gs, tabs));
+    } else {
+        fmt::format_to(std::back_inserter(buf), "{}", this->typeExpr.toStringWithTabs(gs, tabs));
+    }
```

Comment:
ooc, should we keep it showing `todo` anyways?

or change the printing to show all three fields?

I can imagine getting super confused trying to debug something by seeing `T.let(x, Integer)` in two different snapshots and not realizing that they're subtly different because one of them is a constant lit and the other one is a TypePtr.

Reply:
I was trying to avoid having a bunch of `T.let(x, <todo>)` in the changes for exp files (e.g. in rewriter or namer) in later branches, which didn't seem super-helpful.  But I can see confusion arising if you see the same thing being printed at two different times and there are actually two different underlying things (e.g. debugging resolver or something).

My opinion is that if we're going to print three things, the printing should be changed to make it obviously not a send, something like `<cast:let>(x, $TYPE, $TYPE_EXPR)`.  WDYT?

Reply:
wfm